doc/articles/features/using-skia-hosting-native-controls.md

```diff
@@ -2,38 +2,48 @@
+> [!NOTE]
+> As of Uno Platform 6.0, running on macOS does not yet support native controls embedding.
 
-In principle, the native control hosting places the native control in an overlay on top of the Skia control which renders the whole app's UI. The native control is then rendered by the native windowing system and is not part of the Uno Platform visual tree. That control cannot be styled by Uno Platform styles, and Uno Platform controls cannot be placed on top of native controls.
+In an Uno app with a Skia renderer, i.e. using `net9.0-desktop` or adding `SkiaRenderer` to the `UnoFeatures` MSBuild property, you can embed native controls in your Skia app. This is useful if you want to use a native control for a specific task, for instance, to integrate an existing WPF control.
 
-> [!NOTE]
-> As of Uno Platform 5.2, X11 and macOS targets does not yet support native controls embedding.
+Each target platform has its own idea of a native element.
+
+| Platform                        | Native element                                 | Description                                  |
+|---------------------------------|------------------------------------------------|----------------------------------------------|
+| Skia Desktop (Win32)            | Uno.UI.NativeElementHosting.Win32NativeWindow  | A native Windows window with a unique `Hwnd` |
+| Skia Desktop (WPF)              | System.Windows.UIElement                       | A WPF control                                |
+| Skia Desktop (X11)              | Uno.UI.NativeElementHosting.X11NativeWindow    | A native X11 window with a unique `XID`      |
+| WebAssembly with `SkiaRenderer` | Uno.UI.NativeElementHosting.BrowserHtmlElement | An HTML element with a unique `id`.          |
+| Android with `SkiaRenderer`     | Android.Views.View                             | An Android view.                             |
+| Apple UIKit with `SkiaRenderer` | UIKit.UIView                                   | An UIKit view.                               |
+
+The app developer is responsible for creating the native element and internal checks make sure that only a supported native element on the running platform is used.
 
-## Using embedded native controls (Skia.Gtk legacy)
+## Using embedded native controls
 
-To embedded native controls in your Skia heads, you will need to set the native control as `Content` of a `ContentControl`, either via code or XAML.
+To embed a native element, you will need to set the native control as `Content` of a `ContentControl`, either via code or XAML. On desktop platforms, it's often more straightforward to do this via code since the parameters for creating the native element are not known ahead of time. For example, on Win32, you need to create a native Windows window first, get its `Hwnd` and then create a `Win32NativeWindow` instance with that `Hwnd` value.
```

Comment:
```suggestion
To embed a native element, you will need to set the native control as `Content` of a `ContentControl`, either via code or XAML. On desktop platforms, it's often more straightforward to embed a control via code since the parameters for creating the native element are not known ahead of time. For example, on Win32, you need to create a native Windows window first, get its `Hwnd` and then create a `Win32NativeWindow` instance with that `Hwnd` value.
```